packages/r/rust/xmake.lua

```diff
@@ -0,0 +1,47 @@
+package("rust")
+    set_kind("toolchain")
+    set_homepage("https://rust-lang.org")
+
+    if is_host("windows") then
+        add_urls("https://static.rust-lang.org/rustup/archive/1.28.1/x86_64-pc-windows-msvc/rustup-init.exe")
+
+        add_versions("1.86.0", "7b83039a1b9305b0c50f23b2e2f03319b8d7859b28106e49ba82c06d81289df6")
+    else
+        add_urls("https://raw.githubusercontent.com/rust-lang/rustup/refs/tags/1.28.1/rustup-init.sh")
+
+        add_versions("1.86.0", "b25b33de9e5678e976905db7f21b42a58fb124dd098b35a962f963734b790a9b")
+    end
+
+    if is_host("linux") then
+        add_extsources("apt::rustup")
+        add_extsources("pacman::rustup")
+    elseif is_host("macosx") then
+        add_extsources("brew::rustup")
+    end
+
+    on_install(function (package)
+        local cachedir = package:cachedir()
```

Comment:
do use cachedir, the currect directory is in the cachedir.

so you can use the currect relative directory directly.